### ISSUE TITLE
variable-for-property rule should allow reserved words

### ISSUE THREAD

**Issue by jelmerdemaat** [sasstools/sass-lint#571]
**Issue**
The rule `variable-for-property` reports reserved words as being non-variables (which is technically true of course). This means it's impossible to use words like `transparent` or `none`.

**Possible solution**
Make a list of reserved words that are allowed when using `variable-for-property` (equal to [scss-lint](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#variableforproperty), see [this line](https://github.com/brigade/scss-lint/blob/e9a8e243b43bc4fe30a919ebea2e413087546a95/lib/scss_lint/linter/variable_for_property.rb#L6)).

**Example**
This can be tested with the following config:

```yml
options:
  merge-default-rules: false
  formatter: stylish
rules:
  variable-for-property:
    - 2
    -
      properties:
        - background
```

And this CSS:

```css
.test {
  background: inherit;
  background: initial;
  background: transparent;
  background: none;
}
```

Which generates this errors when using sass-lint v1.5.1:

```
test.scss
  2:3  error  Values for properties of type 'background' may only be variables  variable-for-property
  3:3  error  Values for properties of type 'background' may only be variables  variable-for-property
  4:3  error  Values for properties of type 'background' may only be variables  variable-for-property
  5:3  error  Values for properties of type 'background' may only be variables  variable-for-property

✖ 4 problems (4 errors, 0 warnings)
```

**Comment by jelmerdemaat** [comment]
To add, I think this list would include:

* inherit
* initial
* transparent
* none
* currentColor

**Comment by jelmerdemaat** [comment]
In technical terms I would propose the following:

1. Adding a property `whitelist` to the defaults of this rule, which is an array of whitelisted words.
2. When checking if the declaration value is a variable, also check if `whitelist.indexOf(value) === -1`
3. Write a test to check this. :)

If you'd like I could make some time for this the coming days!

**Comment by DanPurdy** [comment]
I think in general those words should just be whitelisted by default and should never form the part of any checks or even subject to this rule. I checked the expected behaviour in scss-lint too so i think it makes sense to keep it like this. Either way I've put a PR to solve this on our updated AST branch which means it should come out with 1.6 which I would really hope to get pushed out this week!

**Comment by jelmerdemaat** [comment]
That would be great, many thanks!

**Comment by DanPurdy** [comment]
@jelmerdemaat This is fixed and available in sass-lint 1.6.0 👍

**Comment by jelmerdemaat** [comment]
Awesome! :+1:

**Comment by nicolas-goudry** [comment]
@DanPurdy Hello, I wonder why this issue didn't add a whitelist option to `variable-for-property` rule? Currently, I'm diging through our codebase to remove all arbitrary fixed `margin` and `padding` and we can sometime use `auto` or `0` which are IMO non-sense to set as `$auto-spacing` or `$no-spacing` variables…

Is it possible to allow users to extend the default whitelist like this?

```yaml
variable-for-property:
  - 1
  -
    properties:
      - 'margin'
      - 'padding'
    whitelist-values:
      - 'auto'
      - '0'
```

If you wish me to open a new issue, I'll do it, just thought it has to be here cause it's clearly related!

I also can work on a PR if you don't have time to dig into this issue right now :)

Thanks!
